### PR TITLE
small refactor handle_music

### PR DESCRIPTION
**Pull request by Salanto**
Refactor handle_music to remove duplicate code

### CODE REVIEW

**Comment by Crystalwarrior** [comment]
Can you make sure this doesn't break anything? Plz test everything just to double check.

**Comment by Salanto** [comment]
From what I can tell, after testing both playing music without showname, with showname, using the /play command on both Akash and Tsuserver3 master, the behaviour matches the currently established one.

File: src/courtroom.cpp [hunk 1]
```diff
@@ -3677,76 +3677,42 @@ void Courtroom::handle_song(QStringList *p_contents)
...
-      else if (f_song.startsWith("http"))
-        ui_music_name->setText(tr("[STREAM] %1").arg(f_song_clear));
-      else
-        ui_music_name->setText(tr("[MISSING] %1").arg(f_song_clear));
-    }
+    effect_flags = p_contents->at(5).toInt();
   }
-  else {
+
+  bool is_stop = (f_song == "~stop.mp3");
```
Comment (in1tiate) [inline src/courtroom.cpp:3707]: this is a nitpick but maybe define `is_stop` at the top with the rest of the variables

File: src/courtroom.cpp [hunk 2]
```diff
@@ -3756,19 +3722,21 @@ void Courtroom::handle_song(QStringList *p_contents)
...
-        else if (f_song.startsWith("http"))
-          ui_music_name->setText(tr("[STREAM] %1").arg(f_song_clear));
-        else
-          ui_music_name->setText(tr("[MISSING] %1").arg(f_song_clear));
-      }
     }
   }
+
+  music_player->play(f_song, channel, looping, effect_flags);
+  if(is_stop) {
```
Comment (in1tiate) [inline src/courtroom.cpp:3729]:
```suggestion
  if (is_stop) {
```

**Review by in1tiate — CHANGES_REQUESTED** [review]
It all looks good, though if possible I'd like to see some checks added for all these `toInt()` calls. Looks like if we get sent a malformed MC packet right now we'll just crash.

File: src/courtroom.cpp (hunk 1 above)
Comment (Salanto) [reply src/courtroom.cpp:3707]: Can't do as f_song is declared after the defaults are set. I would have to nest it in the middle of reading data, which reduces visibility.

File: src/courtroom.cpp [hunk 3]
```diff
@@ -3677,76 +3677,50 @@ void Courtroom::handle_song(QStringList *p_contents)
   if (f_contents.size() < 2)
     return;
 
+  bool ok, ok2, ok3; //CharID,Channel, Effect
```
Comment (oldmud0) [inline src/courtroom.cpp:3680]: don't do this

File: src/courtroom.cpp [hunk 4]
```diff
@@ -3677,76 +3677,50 @@ void Courtroom::handle_song(QStringList *p_contents)
...
-  else {
+  if (p_contents->length() > 5) {
+    // Flags provided to us by server such as Fade In, Fade Out, Sync Pos etc.
+    effect_flags = p_contents->at(5).toInt(&ok3);
+    if (!ok3)
+      return;
+  }
+
+  bool is_stop = (f_song == "~stop.mp3");
+  if (!(n_char < 0) && !(n_char >= char_list.size())) {
```
Comment (oldmud0) [inline src/courtroom.cpp:3716]: what?
```suggestion
  if (n_char > 0 && n_char < char_list.size()) {
```
oh, you big brain...

**Review by oldmud0 — CHANGES_REQUESTED** [review]
just some style issues but otherwise looks ok, you killed a lot of duplicate code

**Comment by Crystalwarrior** [comment]
Would rather this be merged after 2.9.1, this doesn't fix anything and any refractor comes with a substantial risk of new bugs

File: src/courtroom.cpp (hunk 4 above)
Comment (Salanto) [reply src/courtroom.cpp]: Applied suggestion in most recent commit.

File: src/courtroom.cpp (hunk 3 above)
Comment (Salanto) [reply src/courtroom.cpp]: Un-did it.

**Review by in1tiate — APPROVED** [review]
looks good to me, good refactor

**Review by in1tiate — APPROVED** [review]
meant to approve this oops

**Pull request merged**